### PR TITLE
SDL2 port

### PR DESCRIPTION
**Pull request by jmmaunus**
This is a working port of AC from SDL to SDL2. I've tested it under both Linux and Windows, and it seems to run without problems, but some glitches may still remain and I'd like to involve more developers&testers. The *nix Makefile and Code::Blocks project files have been updated and should work, but I haven't touched the vcpp and xcode files.

I won't do a "why SDL2" speech here, but I'll just say that SDL2 is supported and under continuous development (while SDL1 was deprecated years ago) and has a bunch of cool features you can check out.

### CODE REVIEW

**Comment by jmmaunus** [comment]
Don't mind travis erroring for now; it happens because it fails to install SDL2 libraries. I'm trying to come up with a solution, but since Ubuntu 12.04 doesn't have any SDL2 libraries available and there doesn't seem to be any backports of the dev packages, that might take a bit.

**Comment by jmmaunus** [comment]
Travis is now correctly set up for SDL2 and AC passes the build. Huzzah. Go me.

**Comment by ac-stef** [comment]
Since we can only use one SDL version at a time, we need a working build on all supported platforms - and without a competent OSX dev at hand, this may prove difficult.
Also, I'm not yet convinced, that we actually need SDL2, since "being newer" is not reason enough.
While anyone starting a new project would certainly choose SDL2, this game has now 10 years of experience with SDL1. Also, SDL1 can now rightfully be called "stable" while SDL2 is still kind of "new".
In short: I think, "why SDL2?" is a valid question.
I think, it's good, that there's now a branch with SDL2 available - mostly, because I wonder, if it would help with some problems on OSX that we're having. But again, without a dev for that platform...

Well, let's hope, that enough people will spend some time testing this branch...

btw - why is the branch called "gl3"?
 :smiley:

**Comment by ac-stef** [comment]
FYI, the way we usually use our forks and the main repo, it's completely okay to rebase your branches. So far, no one has ever based own branches on branches of forks of AC. In fact, some devs (including me) use their forked repo to build and clean up features (and their commits) until they're ready, at which point the branch will be rebased a last time and then merged fast-forward into the main repo. This leaves the main repo as a mostly linear string of commits without too many "oops fix commits" in it. We try to have only well-compiling (though not on all platforms, usually) and fully functional commits in the main branch.
In short: feel free to rebase this branch instead of merging every change on "next" into it. The general git rule of "never rebase anything that has been pushed to a public repo" doesn't make much sense for AC's forked repos. :smiley:

**Comment by jmmaunus** [comment]
I don't think "why SDL2" is a bad question at all, I was just too lazy to address it properly (that's a perfectly good reason, right?)

As for the name, I was planning to redo the engine for OpenGL 3 core (not replacing the old one but having it as optional, of course), and since it doesn't seem possible to create a core context with SDL1, I thought I might as well take it to SDL2. I've poked around the code enough to say that that's probably far too big an undertaking (given how much AC relies on now deprecated functionality, that would require a total rewrite of nearly all rendering code and maintaining both engines in parallel), so whatever improvements I'll make to the rendering code will most likely be compatible with immediate mode OGL (which wouldn't need SDL2 anyway), but since I had already started I thought I might as well go all the way. It didn't occur to me to rename it...

I'll keep working on (and rebasing! ;)) this branch for now at least.

**Comment by aurhat** [comment]
This cubescript debug is annoying (I can't disable it in AC, I set scriptdebug to 0)...

**Comment by jmmaunus** [comment]
I've created an 'sdl2' branch just for SDL2 - you'll probably want to check that out instead as it doesn't have my terrible commit history (and a rebase bungle). I'm planning to continue my SDL2 work there.

**Comment by RonaldReagan** [comment]
I am on your sdl2 branch, got it compiling and "working" on OSX. However getting some display errors. Is it just me?

Getting something like this:
20151229_07 33 49_ac_avenue_tdm

Haven't done any debug, but no errors in the log, missing text and compass rose on screen (as well as other things).



And this is my console log.
```
writing to file: /Users/Glass/Library/Application Support/assaultcube_1.2r/clientlog.txt
init: sdl (2.0.4)
SDL: compiled version 2.0.4, linked version 2.0.3
init: net (1.3.6)
init: world (1202)
init: video: sdl
init: video: mode
init: video: misc
init: gl
Renderer: AMD Radeon HD 6490M OpenGL Engine (ATI Technologies Inc.)
Driver: 2.1 ATI-1.32.25
init: console
init: sound
Audio devices: Built-in Output
Sound: Built-in Output / Software (Apple Computer Inc.)
Driver: 1.1
init: cfg
loaded 1 xmaps
init: models
init: docs
init: localconnect
read map packages/maps/official/ac_avenue.cgz rev 21 (7 milliseconds)
ac_avenue by jiba
loaded textures (2680 milliseconds)
loaded mapmodels (3555 milliseconds)
loaded mapsounds (992 milliseconds)
game mode is "TDM"
init: mainloop
resolving hostname us.ac-akimbo.net succeeded (104.28.10.74)
resolving hostname de.ac-akimbo.net succeeded (104.28.11.74)
resolving hostname packages.ac-akimbo.net succeeded (104.28.11.74)
http head us.ac-akimbo.net:80/updates.txt response time: 450, status: 404, contentlength: 0
http head packages.ac-akimbo.net:80/updates.txt response time: 601, status: 404, contentlength: 0
http head de.ac-akimbo.net:80/updates.txt response time: 603, status: 404, contentlength: 0
successfully pinged 3 media servers, 0 failures, 0 disabled
```

**Comment by jmmaunus** [comment]
Is text in menus missing as well? Is anything displayed in the bigger console mode (F11)? I can't reproduce this behaviour on any of my systems, but I have a feeling it's a texture issue.

**Comment by jmmaunus** [comment]
I've created a new pull request on the proper branch so I'm closing this (can't remember why I didn't do that in the first place), we can continue the discussion there.
